tests/Integration/ProducerTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+namespace Tests\Integration;
+
+use Illuminate\Support\Facades\Log;
+use Metamorphosis\Facades\Metamorphosis;
+use Tests\Integration\Dummies\MessageConsumer;
+use Tests\Integration\Dummies\MessageProducer;
+use Tests\LaravelTestCase;
+
+class ProducerTest extends LaravelTestCase
+{
+    /**
+     * @var string
+     */
+    protected $highLevelMessage;
+
+    /**
+     * @var string
+     */
+    protected $firstLowLevelMessage;
+
+    /**
+     * @var string
+     */
+    protected $secondLowLevelMessage;
+
+    protected function setUp()
+    {
+        parent::setUp();
+        $this->withoutAuthentication();
+    }
+
+    public function testShouldRunAProducerAndReceiveMessagesWithAHighLevelConsumer(): void
+    {
+        // Given That I
+        $this->haveAConsumerHandlerConfigured();
+        $this->haveSomeRandomMessagesProduced();
+
+        // I Expect That
+        $this->myMessagesHaveBeenLogged();
+
+        // When I
+        $this->runTheConsumer();
+    }
+
+    public function testShouldRunAProducerAndReceiveMessagesWithALowLevelConsumer(): void
+    {
+        // Given That I
+        $this->haveSomeMessagesToBeSent();
+        $this->haveALowLevelConsumerConfigured();
+        $this->haveTwoMessagesProducedBefore();
```

Comment:
you could say something like `haveFourProducedMessages` and merge this method and the method below